e2e-tests/test/hardhat-apis.test.ts

```diff
@@ -118,6 +118,22 @@ describe("hardhat_setCode", function () {
     expect(BigNumber.from(result).toNumber()).to.eq(5);
   });
 
+  it("Should reject invalid code", async function () {
+    const action = async () => {
+      const wallet = new Wallet(RichAccounts[0].PrivateKey);
+      const deployer = new Deployer(hre, wallet);
+
+      const address = "0x1000000000000000000000000000000000001111";
+      const artifact = await deployer.loadArtifact("Return5");
+      const contractCode = [...ethers.utils.arrayify(artifact.deployedBytecode)];
+      const shortCode = contractCode.slice(0, contractCode.length - 1);
+
+      await provider.send("hardhat_setCode", [address, shortCode]);
+    };
+
+    await expectThrowsAsync(action);
```

Comment:
This requires a second parameter, the `errorMessage` string to be a substring in the expected error to be thrown

Reply:
```suggestion
    await expectThrowsAsync(action, "bytes must be divisible by 32");
```

Reply:
Doesn't look like I chose the correct error message 😅

Reply:
The error message is correct - it's just that the server looses it when converting the error through `Web3Error::InternalError`... On the other hand, there is no reason it should convert errors through `Web3Error::InternalError` (except it had been implemented that way) - with https://github.com/matter-labs/era-test-node/pull/293/commits/8b7b5e1639a0c0b4bdc1eb7db833e011302cffff , the test passes unmodified. This change should probably be applied more widely... :-)